code/game/objects/items/weapons/shields.dm

```diff
@@ -222,24 +340,53 @@
 			user.visible_message(SPAN_DANGER("\The [user] blocks [attack_text] with \the [src]!"))
 			return PROJECTILE_STOPPED
 
+	return FALSE
+
+/obj/item/shield/energy/handle_damage(var/mob/user, var/damage)
+	if(shield_health == 0)
+		return
 
-/obj/item/shield/energy/get_block_chance(mob/user, damage, atom/damage_source = null, mob/attacker = null)
-	if(isprojectile(damage_source))
-		if((is_sharp(damage_source) && damage > 10) || isbeam(damage_source))
-			return (base_block_chance - round(damage / 3))
-	return base_block_chance
+	spark(user.loc, 5)
+	shield_health = clamp(shield_health - damage, 0, max_shield_health)
+
+	if(!(datum_flags & DF_ISPROCESSING))
+		START_PROCESSING(SSprocessing, src)
```

Comment:
Why start processing in here, instead of when the shield is turned on? And stop processing when it's turned off?

Reply:
I don't want it to process unnecessarily while it's active, when it only needs to process when it's drained and then stop when it's fully powered

Reply:
I would have bet to have seen the battery were draining by just being on but I guess I was cooked yesterday and was thinking of the portable belt shield instead